ambry-store/src/main/java/com.github.ambry.store/IndexSegment.java

```diff
@@ -341,11 +350,11 @@ public void addEntry(IndexEntry entry, long fileEndOffset) throws StoreException
       }
       logger.trace("IndexSegment {} inserting key - {} value - offset {} size {} ttl {} "
               + "originalMessageOffset {} fileEndOffset {}", indexFile.getAbsolutePath(), entry.getKey(),
-          entry.getValue().getOffset(), entry.getValue().getSize(), entry.getValue().getTimeToLiveInMs(),
+          entry.getValue().getOffset(), entry.getValue().getSize(), entry.getValue().getExpiresAtMs(),
           entry.getValue().getOriginalMessageOffset(), fileEndOffset);
       if (index.put(entry.getKey(), entry.getValue()) == null) {
         numberOfItems.incrementAndGet();
-        sizeWritten.addAndGet(entry.getKey().sizeInBytes() + IndexValue.Index_Value_Size_In_Bytes);
+        sizeWritten.addAndGet(entry.getKey().sizeInBytes() + entry.getValue().getBytes().capacity());
```

Comment:
A given `IndexSegment` in theory is *not* designed to handle any value size. I think you meant across segments. Within a segment it has to be the same for the binary search to work correctly. We do ensure that a rollover happens whenever the value size changes, but that code is not well written right now (may not even be correct). If we set about making this change I think you should get rid of this static variable (`Index_Value_Size_In_Bytes`) altogether. What does it really represent anyway? It seems like this should be a segment specific variable.